Toolkit/Public/Get-RscAzureNativeVm.ps1

```diff
@@ -0,0 +1,95 @@
+#Requires -Version 3
+function Get-RscAzureNativeVm {
+    <#
+    .SYNOPSIS
+    Retrieves RscAzureNativeVm objects protected by Rubrik Security Cloud
+
+    .DESCRIPTION
+    This cmdlet uses the GQL query 'azureNativeVirtualMachines' to retrieve a list of VMs with a predetermined set of properties.
+
+    .LINK
+    Schema reference:
+    https://rubrikinc.github.io/rubrik-api-documentation/schema/reference
+
+    .EXAMPLE
+    # Get all
+    Get-RscAzureNativeVm
+
+    .EXAMPLE
+    # Get object with specific name
+    Get-RscAzureNativeVm -Name "jake-001"
+
+    .EXAMPLE
+    # Get objects by specifying part of a name
+    Get-RscAzureNativeVm -Name "*jake*"
+    #>
+
+    [CmdletBinding(
+        DefaultParameterSetName = "Name"
+    )]
+    Param(
+        [Parameter(
+            Mandatory = $false,
+            ParameterSetName = "Id"
+        )]
+        [String]$Id,
+        [Parameter(
+            Mandatory = $false,
+            ParameterSetName = "Name"
+        )]
+        [String]$Name,
+        [Parameter(
+            Mandatory = $false,
+            ValueFromPipeline = $true,
+            ParameterSetName = "Name"
+        )]
+        [RubrikSecurityCloud.Types.GlobalSlaReply]$Sla
+    )
+    
+    Process {
+
+       # The query is different for getting a single object by ID.
+        if ($Id) {
+            $query = New-RscQuery -GqlQuery azureNativeVirtualMachine
+            $query.var.filter = @()
+            $query.Var.fid = $Id
+
+            $result = Invoke-Rsc -Query $query
+            $result
+        } else {
+            $query = New-RscQuery -GqlQuery azureNativeVirtualMachines
+            $query.var.filter = @()
+
+            if ($Name) {
+                $nameFilter = New-Object -TypeName RubrikSecurityCloud.Types.Filter
+                # Regex filter doesn't work in the API right now, but we're going to play pretend. 
+                # With real Regex, users could search for VMs that start with the letter A if they wanted.
+                if ($name.Contains("*")) {
+                    $name.Replace("*",'')
+                    $nameFilter.Field = [RubrikSecurityCloud.Types.HierarchyFilterField]::REGEX
+                    $nameFilter.texts = $Name.Replace("*",'')
+                } else {
+                    $nameFilter.Field = [RubrikSecurityCloud.Types.HierarchyFilterField]::NAME_EXACT_MATCH
+                    $nameFilter.texts = $Name
+                }
+                $query.var.filter += $nameFilter
+            }
+    
+            if ($Sla) {
```

Comment:
$Name and $Sla should be mutually exclusive

Reply:
because we can only filter on name substring, not an exact match, we need to be able to filter on sla at the same time